src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -160,7 +160,7 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 		let result = `${value.substring(0, this.cursorIndex)}|`;
 		if (this.ghostTextIndex !== -1) {
 			result += `${value.substring(this.cursorIndex, this.ghostTextIndex)}[`;
-			result += `${value.substring(this.ghostTextIndex)}]`;
+			result += `${value.trimEnd().substring(this.ghostTextIndex)}]`;
```

Comment:
If the whitespace is after `ghostTextIndex`, we can clear this up just after we detect the ghost text instead. That way this problem shouldn't just get fixed in `getCombinedString`. What we cannot do is trim if `ghostTextIndex` is -1 though since the command line may have trailing spaces.